ticdc/ticdc-simple-protocol.md

```diff
@@ -0,0 +1,714 @@
+| ------------- | ------- | --------------------------------------------------------- |
+| `version`     | Number  | The version number of the protocol, which is currently `1`. |
+| `type`        | String  | The `BOOTSTRAP` event type.                               |
+| `commitTs`    | Number  | The `commitTs` of the `BOOTSTRAP` is `0`. Because it is generated internally by TiCDC, its `commitTs` is meaningless. |
+| `buildTs`     | Number  | The UNIX timestamp when the message is successfully encoded within TiCDC. |
+| `tableSchema` | Object  | The schema information of the table. For more information, see [TableSchema definition](#tableschema-definition). |
+
+## Message generation and sending rules
+
+### DDL
+
+- Generation time: TiCDC sends a DDL event after all transactions before this DDL event have been sent.
+- Destination: TiCDC sends DDL events to all partitions of the corresponding topic.
+
+### DML
+
+- Generation time: TiCDC sends DML events in the order of the `commitTs` of the transaction.
+- Destination: TiCDC sends DDL events to the corresponding partition of the corresponding topic according to the user-configured dispatch rules.
+
+### WATERMARK
+
+- Generation time: TiCDC sends `WATERMARK` events periodically to mark the replication progress of a changefeed. The current interval is 1 second.
+- Destination: TiCDC sends `WATERMARK` events to all partitions of the corresponding topic.
+
+### BOOTSTRAP
+
+- Generation time:
+    - After creating a new changefeed, before the first DML event of a table is sent, TiCDC sends a `BOOTSTRAP` event to the downstream to build the table schema.
+    - Additionally, TiCDC sends `BOOTSTRAP` events periodically to allow newly joined consumers to build the table schema. The default interval is 120 seconds or every 10000 messages. You can adjust the sending interval by configuring the `send-bootstrap-interval-in-sec` and `send-bootstrap-in-msg-count` parameters in the `sink` configuration.
+    - If a table does not receive any new DML messages within 30 minutes, the table is considered inactive. TiCDC stops sending `BOOTSTRAP` events for the table until new DML events are received.
+- Destination: By default, TiCDC sends `BOOTSTRAP` events to all partitions of the corresponding topic. You can adjust the sending strategy by configuring the `send-bootstrap-to-all-partition` parameter in the sink configuration.
+
+## Message consumption methods
+
+Because the TiCDC Simple protocol does not include the schema information of the table when sending DML messages, the downstream needs to receive DDL or BOOTSTRAP messages and cache the schema information of the table when consuming DML messages. When receiving a DML message, the downstream can obtain the corresponding table schema information from the cache through the `table` name and `schemaVersion` field in the DML message, and then correctly consume the DML message.
```

Comment:
```suggestion
Because the TiCDC Simple protocol does not include the schema information of the table when sending a DML message, the downstream needs to receive the DDL or BOOTSTRAP message and cache the schema information of the table before consuming a DML message. When receiving a DML message, the downstream obtains the corresponding table schema information from the cache by searching the `table` name and `schemaVersion` fields of the DML message, and then correctly consumes the DML message.
```